src/mpi-base.jl

```diff
@@ -5,11 +5,73 @@ typealias MPIDatatype Union{Char,
 typealias MPIBuffertype{T} Union{Ptr{T}, Array{T}, Ref{T}}
 
 # Define a function mpitype(T) that returns the MPI datatype code for
-# a given type T. The dictonary is defined in __init__ so the module
-# can be precompiled
+# a given type T. In the case the the type does not exist, it is created and
+# then returned. The dictonary is defined in __init__ so the module can be
+# precompiled
+
+# accessor and creation function for getting MPI datatypes
+function mpitype{T}(::Type{T})
+
+    if haskey(mpitype_dict, T)  # if the datatype already exists
```

Comment:
I was speaking of pointers, in the C sense. If you define
```C
double x = 1;
const double y = 2;
int f() { return 3; }
```
and then take the pointers `&x`, `&y`, or `&f`, they will often be the same if MPI is running the same executable on different processes (apart from ASLR etc.). In this case, it does make sense to pass pointers between processes.